pandas/parser.pyx

```diff
@@ -548,6 +555,29 @@ cdef class TextReader:
     def set_error_bad_lines(self, int status):
         self.parser.error_bad_lines = status
 
+    def _set_quoting(self, quote_char, quoting):
+        if not isinstance(quoting, int):
+            raise TypeError('"quoting" must be an integer')
+
+        if not 0 <= quoting <= 3:
```

Comment:
Fair enough.  It's a little nicer than the two checks that I have current (e.g. someone for whatever reason could pass in `np.int32(2)` and that would raise).